### PR TITLE
Adding RSpec environment for testing 

### PR DESCRIPTION
**Pull request by angiecstn**
PT Chore: https://www.pivotaltracker.com/story/show/160587751 

Proposed Changes:

Setting up the test environment for RSpec.

Added gems:

gem 'factory_bot_rails'
gem 'pry-rails'
gem 'rspec-rails'
gem 'shoulda-matchers'

Configured shoulda-matchers
Disabled generation of helpers, javascripts, css, and view, helper, routing and controller specs to avoid too many files being created

added --format documentation in RSpec

We learned how to set up the test environment for RSpec following yesterdays implementation.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -18,6 +18,11 @@ gem 'uglifier', '>= 1.3.0'
 
 group :development do
   gem 'pry-byebug'
```
Comment (faraznaeem) [inline Gemfile]: you can skip `pry-byebug`

File: Gemfile [hunk 2]
```diff
@@ -18,6 +18,11 @@ gem 'uglifier', '>= 1.3.0'
...
+  gem 'factory_bot_rails'
+  gem 'pry-rails'
+  gem 'rspec-rails'
+  gem 'shoulda-matchers'
+
   gem 'spring'
```
Comment (faraznaeem) [inline Gemfile]: remove empty lines

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  # Filter lines from Rails gems in backtraces.
+  config.filter_rails_from_backtrace!
+  # arbitrary gems may also be filtered via:
+  # config.filter_gems_from_backtrace("gem name")
+end
+
+Shoulda::Matchers.configure do |config|
+  config.integrate do |with|
+    with.test_framework :rspec
+    with.library :rails
```
Comment (faraznaeem) [inline spec/rails_helper.rb:25]: Clean this file out from comments

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  # the seed, which is printed after each run.
+  #     --seed 1234
+  config.order = :random
+
+  # Seed global randomization in this process using the `--seed` CLI option.
+  # Setting this allows you to use `--seed` to deterministically reproduce
+  # test failures related to randomization by passing the same `--seed` value
+  # as the one that triggered the failure.
+  Kernel.srand config.seed
+=end
```
Comment (faraznaeem) [inline spec/spec_helper.rb]: you can clean out this file too

**Comment by faraznaeem** [comment]
Please add the user story to the PR description and the link to the PT or waffle issue

**Review by faraznaeem — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (AlbertStjarne) [reply Gemfile]: fixed

**Comment by AlbertStjarne** [comment]
All comments on code changes taken care of.

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**